core/src/main/java/org/arquillian/smart/testing/TestSelection.java

```diff
@@ -14,23 +16,28 @@
 
     private final String className;
 
-    private final Collection<String> types; // TODO or strategy instead (misleading name)
+    private final Collection<String> appliedStrategies;
 
-    public TestSelection(Path location, String... type) {
-        this(new ClassNameExtractor().extractFullyQualifiedName(location), type);
+    private final List<String> testMethodNames;
+
+    public static final TestSelection NOT_MATCHED = new TestSelection("");
+
+    public TestSelection(Path location, String... appliedStrategies) {
+        this(new ClassNameExtractor().extractFullyQualifiedName(location), appliedStrategies);
     }
 
-    public TestSelection(String className, String ... type) {
+    public TestSelection(String className, String ... appliedStrategies) {
         this.className = className;
-        this.types = new LinkedHashSet<>(asList(type));
+        this.appliedStrategies = new LinkedHashSet<>(asList(appliedStrategies));
+        this.testMethodNames = new ArrayList<>();
     }
 
     public String getClassName() {
         return className;
     }
 
-    public Collection<String> getTypes() {
-        return types; // TODO should we return clone to avoid manipulation?
+    public Collection<String> getAppliedStrategies() {
+        return appliedStrategies; // TODO should we return clone to avoid manipulation?
```

Comment:
So what do we do? (`TODO`)

Reply:
I haven't decided yet :-D. 
To be honest, I don't see any potential use case when it could cause some problems. We don't expose this instance of `TestSelection` class to anyone, we just take it from the strategies and merge them internally. 
Correct me if I'm wrong.

Reply:
WDYT?